mechanism2d/robot.py

```diff
@@ -0,0 +1,57 @@
+import wpilib
+
+# This sample program shows how to use Mechanism2d - a visual representation of arms, elevators,
+# and other mechanisms on dashboards; driven by a node-based API.
+
+# <p>Ligaments are based on other ligaments or roots, and roots are contained in the base
+# Mechanism2d object.
+
+
+class MyRobot(wpilib.TimedRobot):
+
+    kMetersPerPulse = 0.01
+    kElevatorMinimumLength = 0.5
+
+    def robotInit(self):
+        self.elevatorMotor = wpilib.PWMSparkMax(0)
+        self.wristMotor = wpilib.PWMSparkMax(1)
+        self.wristPot = wpilib.AnalogPotentiometer(1, 90)
+        self.elevatorEncoder = wpilib.Encoder(0, 1)
+        self.joystick = wpilib.Joystick(0)
+
+        self.elevator = wpilib.MechanismLigament2d
+        self.wrist = wpilib.MechanismLigament2d
```

Comment:
I'm confused as to what's happening here. Shouldn't the right hand side be constructing a `MechanismLigament2d`, rather than a reference to the type?

Reply:
It doesn't have a __init__() function, it is referenced rather than constructed in the Java examples too. I'm not sure why they decided to do it this way, maybe it's something to do with how Java -which i have no experience in-, handles variables? I deleted those lines and it seems to be working fine.

Reply:
Doing the equivalent of this in Java would be a compile-time type error, so I'm not sure I understand what you're saying.

Reply:
[Lines 36&37](https://github.com/wpilibsuite/allwpilib/blob/b53b3526a27877d71883fe944e89577a08f9e0ff/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mechanism2d/Robot.java)

What does those lines do?

Reply:
36 and 37 declare variables in the Java example, which are later assigned to on lines 50/51. That isn't needed in python.